pkg/rancher-desktop/components/ContainerLogs.vue

```diff
@@ -0,0 +1,384 @@
+      default:  false,
+    },
+    namespace: {
+      type:    String,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      isLoading:              true,
+      error:                  null,
+      terminal:               null,
+      fitAddon:               null,
+      searchAddon:            null,
+      streamProcess:          null,
+      resizeHandler:          null,
+      resizeObserver:         null,
+      reconnectAttempts:      0,
+      maxReconnectAttempts:   5,
+      searchDebounceTimer:    null,
+      waitingForInitialLogs:  true,
+      revealTimeout:          null,
+      hasReceivedLogs:        false,
+    };
+  },
+  mounted() {
+    this.initializeLogs();
+  },
+  watch: {
+    containerId: {
+      handler() {
+        if (this.terminal) {
+          this.cleanup();
+        }
+        this.initializeLogs();
+      },
+    },
+  },
+  beforeUnmount() {
+    this.cleanup();
+  },
+  methods: {
+    async initializeLogs() {
+      if (window.ddClient) {
+        await this.startStreaming();
+      }
+    },
+    async initializeTerminal() {
+      this.isLoading = false;
+      await this.$nextTick();
+      if (this.$refs.terminalContainer) {
+        this.terminal = new Terminal({
+          theme: {
+            background:    '#1a1a1a',
+            foreground:    '#e0e0e0',
+            cursor:        '#8be9fd',
+            selection:     'rgba(139, 233, 253, 0.3)',
+            black:         '#000000',
+            red:           '#ff5555',
+            green:         '#50fa7b',
+            yellow:        '#f1fa8c',
+            blue:          '#8be9fd',
+            magenta:       '#ff79c6',
+            cyan:          '#8be9fd',
+            white:         '#f8f8f2',
+            brightBlack:   '#6272a4',
+            brightRed:     '#ff6e6e',
+            brightGreen:   '#69ff94',
+            brightYellow:  '#ffffa5',
+            brightBlue:    '#d6acff',
+            brightMagenta: '#ff92df',
+            brightCyan:    '#a4ffff',
+            brightWhite:   '#ffffff',
+          },
+          fontSize:     14,
+          fontFamily:   '\'Courier New\', \'Monaco\', monospace',
+          cursorBlink:  false,
+          disableStdin: true,
+          convertEol:   true,
+          scrollback:   50000,
+          wordWrap:     true,
+        });
+
+        this.fitAddon = new FitAddon();
+        this.terminal.loadAddon(this.fitAddon);
+
+        this.searchAddon = new SearchAddon();
+        this.terminal.loadAddon(this.searchAddon);
+
+        this.terminal.loadAddon(new WebLinksAddon((event, uri) => {
+          event.preventDefault();
+          window.open(uri, '_blank');
+        }));
+
+        this.terminal.open(this.$refs.terminalContainer);
+
+        await this.$nextTick();
+        this.fitAddon.fit();
+
+        this.terminal.write('\x1b[?25l');
+
+        this.resizeHandler = () => {
+          if (this.fitAddon) {
+            this.fitAddon.fit();
+          }
```

Comment:
```suggestion
          this.fitAddon?.fit();
```
Same a few lines down.